src/relayedTransactionV1Builder.ts

```diff
@@ -0,0 +1,113 @@
+import {Transaction} from "./transaction";
+import {IAddress, INonce} from "./interface";
+import {INetworkConfig} from "./interfaceOfNetwork";
+import {ErrInvalidRelayedV1BuilderArguments} from "./errors";
+import {TransactionPayload} from "./transactionPayload";
+import {ContractFunction, StringValue} from "./smartcontracts";
+import {Address} from "./address";
+import BigNumber from "bignumber.js";
+
+export class RelayedTransactionV1Builder {
+    innerTransaction: Transaction | undefined;
+    relayerAddress: IAddress | undefined;
+    relayerNonce: INonce | undefined;
+    netConfig: INetworkConfig | undefined;
+
+    /**
+     * Sets the inner transaction to be used. It has to be already signed.
+     *
+     * @param {Transaction} transaction The inner transaction to be used
+     */
+    setInnerTransaction(transaction: Transaction): RelayedTransactionV1Builder {
```

Comment:
Could we have received an `ITransaction` here? Please leave as it is otherwise :)

Reply:
no, we cannot use that